examples/dataprocserverless-running-notebooks/composer_input/DAGs/serverless_airflow.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This orchestrates the workflow of running Jupyter Notebooks via PySpark job on a Dataproc cluser
+
+This DAG relies on two Airflow variables
+https://airflow.apache.org/concepts.html#variables
+* gcp_project - Google Cloud Project to use for the Cloud Dataproc cluster.
+* gce_zone - Google Compute Engine zone where Cloud Dataproc cluster should be
+  created.
+"""
+
+import datetime
+import uuid
+
+from airflow import models
+from airflow.models import Variable
+from airflow.providers.google.cloud.operators.dataproc import DataprocCreateBatchOperator
+
+yesterday = datetime.datetime.combine(
+    datetime.datetime.today() - datetime.timedelta(1),
+    datetime.datetime.min.time())
+default_dag_args = {
+    # Setting start date as yesterday starts the DAG immediately when it is
+    # detected in the Cloud Storage bucket.
+    'start_date': yesterday,
+    # To email on failure or retry set 'email' arg to your email and enable
+    # emailing here.
+    'email_on_failure': False,
+    'email_on_retry': False,
+    # If a task fails, retry it once after waiting at least 5 minutes
+    'retries': 1,
+    'retry_delay': datetime.timedelta(minutes=5),
+}
+gcp_project = Variable.get('gcp_project')
+gcs_bucket =Variable.get('gcs_bucket')
+phs_region = Variable.get('phs_region')
+phs = Variable.get('phs')
+
+# Arguments to pass to Cloud Dataproc job.
+# EDIT: input_notebook path, output_notebook path, 
+input_notebook = f"gs://{gcs_bucket}/dataproc_serverless/notebooks/jupyter/spark_notebook.ipynb"
+output_notebook = f"gs://{gcs_bucket}/dataproc_serverless/notebooks/jupyter/output/spark_notebook_output.ipynb"
```

Comment:
output_notebook should be tight with a datetime subpath. Ex: ...rless/notebooks/jupyter/output/<2022_03_21_18_09>/spark....ipynb
Because there is one output per run